### PR TITLE
test: add NIP-22 created_at integration coverage

### LINKED ISSUE #505: test: integration tests for NIP-22 created_at timestamp limits
## Summary

The relay enforces `createdAt.maxPositiveDelta` (default 900s) and `createdAt.maxNegativeDelta` (default 0 = unlimited) from settings to reject events with implausible timestamps. There are no integration tests covering this enforcement.

## What to test

- Event with `created_at` exactly at the current time → accepted
- Event with `created_at` 900 seconds in the future (at the limit) → accepted
- Event with `created_at` 901 seconds in the future (over the limit) → rejected with `invalid:`
- When `maxNegativeDelta` is configured, events older than the limit → rejected
- Events within the negative delta limit → accepted

## Relevant code

- `resources/default-settings.yaml` — `limits.event.createdAt`
- `src/@types/settings.ts`
- `src/handlers/event-message-handler.ts`

## Suggested location

`test/integration/features/nip-22/nip-22.feature`

### PR DESCRIPTION
**Pull request by Priyanshubhartistm**
## Description
Added integration tests for NIP-22 `created_at` timestamp limits as requested in issue #505.

These tests verify that the relay correctly enforces:
- `createdAt.maxPositiveDelta` (default: 900 seconds)
- `createdAt.maxNegativeDelta` (default: 0 = unlimited)

## Related Issue
Closes #505

## Motivation and Context
Currently, there were no integration tests covering the timestamp validation logic for event `created_at` field.  
This PR adds proper Cucumber integration tests to ensure events with implausible timestamps (too far in future or past) are rejected while valid ones are accepted.

## How Has This Been Tested?
- Added new feature file: `test/integration/features/nip-22/nip-22.feature`
- Added corresponding step definitions: `test/integration/features/nip-22/nip-22.feature.ts`
- Tests cover all cases mentioned in the issue:
  - Event at current time → accepted
  - Event at +900s (limit) → accepted
  - Event at +901s → rejected with "invalid"
  - Event older than configured `maxNegativeDelta` → rejected
  - Event within `maxNegativeDelta` → accepted
- Tests were written following the existing integration test patterns in the project.
- After hook properly restores default settings to avoid affecting other tests.

## Types of changes
- [x] New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code follows the code style of this project.
- [x] I have added tests to cover my code changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/handlers/event-message-handler.ts [hunk 1]
```diff
@@ -165,7 +165,7 @@ export class EventMessageHandler implements IMessageHandler {
       limits.createdAt.maxPositiveDelta > 0 &&
       event.created_at > now + limits.createdAt.maxPositiveDelta
     ) {
-      return `rejected: created_at is more than ${limits.createdAt.maxPositiveDelta} seconds in the future`
```
Comment (cameri) [inline src/handlers/event-message-handler.ts:168]: let's please stick with rejected

File: test/integration/features/nip-22/nip-22.feature [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    And created_at limits are set to maxPositiveDelta 900 and maxNegativeDelta 0
+    When Alice drafts a text_note event with content "test event" and created_at 0 seconds from now
+    Then Alice sends their last draft event successfully
+    When Alice subscribes to author Alice
+    Then Alice receives a text_note event from Alice with content "test event"
+
+  Scenario: Event with created_at at positive delta limit is accepted
+    Given someone called Alice
+    And created_at limits are set to maxPositiveDelta 900 and maxNegativeDelta 0
+    When Alice drafts a text_note event with content "test event" and created_at 900 seconds from now
```
Comment (cameri) [inline test/integration/features/nip-22/nip-22.feature]: this is a potential flaky integration test since it sits in the boundary, we can probably remove it since it's testing an edge case

File: test/integration/features/nip-22/nip-22.feature [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    Then Alice sends their last draft event successfully
+    When Alice subscribes to author Alice
+    Then Alice receives a text_note event from Alice with content "test event"
+
+  Scenario: Event with created_at above positive delta limit is rejected
+    Given someone called Alice
+    And created_at limits are set to maxPositiveDelta 900 and maxNegativeDelta 0
+    When Alice drafts a text_note event with content "test event" and created_at 901 seconds from now
```
Comment (cameri) [inline test/integration/features/nip-22/nip-22.feature]: let's bump this to 910 so this test case isn't flaky if node gets slow

**Review by cameri — APPROVED** [review]

**Comment by Priyanshubhartistm** [comment]
@cameri thank you
